cppcache/src/TcpConn.cpp

```diff
@@ -318,14 +237,35 @@ size_t TcpConn::socketOp(TcpConn::SockOp op, char *buff, size_t len,
     return totalsend;
   }
 }
+ssize_t TcpConn::doOperation(const TcpConn::SockOp& op, void* buff,
+                             size_t sendlen, ACE_Time_Value& waitTime,
+                             size_t& readLen) const {
+  if (op == SOCK_READ) {
+    return stream_->recv_n(buff, sendlen, &waitTime, &readLen);
+  } else {
+    return stream_->send_n(buff, sendlen, &waitTime, &readLen);
+  }
+}
 
 //  Return the local port for this TCP connection.
 uint16_t TcpConn::getPort() {
   ACE_INET_Addr localAddr;
-  m_io->get_local_addr(localAddr);
+  stream_->get_local_addr(localAddr);
   return localAddr.get_port_number();
 }
 
+size_t TcpConn::getDefaultChunkSize() {
+  //
+  auto pageSize = boost::interprocess::mapped_region::get_page_size();
+  if (pageSize > 16000000) {
+    return 16000000;
+  } else if (pageSize > 0) {
+    return pageSize + (16000000 / pageSize) * pageSize;
+  }
+
+  return 16000000;
```

Comment:
So line 263 is some misguided attempt to use integer division to our advantage, but I'm pretty sure it's wrong, unless for some reason we know the Geode server is doing the exact same bizarre calculation and we need to replicate it.  What the heck is this used for?

Reply:
Okay, the comment deleted from the original gives it away:
    // Attempt to set chunk size to nearest OS page size	
    // for perf improvement
Spoiler alert, this will NOT improve perf in any perceptible way.

Reply:
Yeah, I didn't want to mess with it. We could through the comment back in but I felt it was an inaccurate statement. Clearly it was picking something at the os page size given the method name it was calling, but the other magic going on didn't. Lacking interest to write a benchmark and test around the change to pull it I just left the code.

Reply:
As for the constant, sure, we could add one but its really outside the scope of the PR to make all the cleanups. Keep in mind while it appears this code is new its actually just moved from the external file to the internal file. GIt just can't see that.

Reply:
I remember now why didn't make a constant. For exactly your second question. Without an answer to that I lacked anything meaningful to name it. A constant `constexpr auto SOME_NUMBER = 16000000` felt more wrong.